### PR TITLE
Document requiring custom syntaxes in shared configs

### LINKED ISSUE #5632: Fix custom syntax module path not properly resolving
> Clearly describe the bug

As per https://github.com/stylelint-scss/stylelint-config-standard-scss/issues/2, custom syntax module path is not properly resolved.

Stylelint tries to resolve custom syntax location ([stylelint/stylelint@`main`/lib/getPostcssResult.js#L95](https://github.com/stylelint/stylelint/blob/main/lib/getPostcssResult.js#L95)) but uses Node `require` algorithm, so it search from that file upwards.

If you have dependancy tree where `postcss-scss` is not in one of the root directories, but is nested inside this config’s `node_modules` (it happens if you have different versions of same package), path resolution fails.

> Which rule, if any, is the bug related to?

None, it’s related to core functionality

> What code is needed to reproduce the bug?

Even empty SCSS file would do.

> What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard-scss"
}
```

But `postcss-scss` shouldn’t be installed inside root `node_modules`.

> Which version of Stylelint are you using?

`14.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`stylelint "**/*.scss"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS custom syntax.

> What did you expect to happen?

Should return at least error/warning related to style code or exit code `0` if everything is OK.

> What actually happened (e.g. what warnings or errors did you get)?

```shell
Cannot resolve custom syntax module "postcss-scss". Check that module "postcss-scss" is available and spelled correctly.

at getCustomSyntax (node_modules/stylelint/lib/getPostcssResult.js:97:10)
at getPostcssResult (node_modules/stylelint/lib/getPostcssResult.js:41:5)
at lintSource (node_modules/stylelint/lib/lintSource.js:76:20)
at Function.standalone [as lint] (node_modules/stylelint/lib/standalone.js:132:26)
```

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Closes #5632

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Comment by jeddy3** [comment]
Do we need coveralls, or can someone with access remove it?

File: docs/developer-guide/syntaxes.md [hunk 1]
```diff
@@ -17,13 +17,17 @@ We recommend creating a shared-config that:
...
-    "at-rule-no-unknown": null,
-    ..
+```js
+module.exports = {
+  {
+    "extends": ["stylelint-config-standard"],
+    "customSyntax": require("postcss-foo"),
+    "rules": {
+      "at-rule-no-unknown": null,
+      ..
```
Comment (ybiquitous) [inline docs/developer-guide/syntaxes.md:27]: [trivial] How about changing to valid JS as below?

```suggestion
      // ..
```

**Review by ybiquitous — APPROVED** [review]
@jeddy3 Thanks! LGTM! 👍🏼

**Comment by clown-helang** [comment]
Has the new version of this issue been released?

**Comment by jeddy3** [comment]
@clown-helang This was a documentation change that's [live on the website](https://stylelint.io/developer-guide/syntaxes) and [shared configs are now following the recommendation](https://github.com/stylelint-scss/stylelint-config-recommended-scss/blob/d77fcac68f819054b501496476d8acd6085cbc4c/index.js#L5), e.g. the SCSS one uses:

```
customSyntax: require('postcss-scss'),
```

The `customSyntax` configuration property has been available [since `14.0.0`](https://stylelint.io/CHANGELOG#1400), which was released a few months ago.

**Pull request merged**